sno/db_util.py

```diff
@@ -0,0 +1,61 @@
+from psycopg2.sql import Identifier, SQL
+
+from . import gpkg
+
+
+# Utilities for dealing with different database drivers.
+
+
+def execute_insert_dict(dbcur, table, record, gpkg_funcs=None, pg_funcs=None):
+    if is_postgis(dbcur):
+        sql = SQL("INSERT INTO {} ({}) VALUES ({});").format(
+            Identifier(table),
+            SQL(",").join([Identifier(k) for k in record]),
+            SQL(",").join(_postgis_placeholders(record, pg_funcs)),
+        )
+    else:
+        sql = f"""
+        INSERT INTO {table}
+            ({','.join([gpkg.ident(k) for k in record])})
+        VALUES
+            ({','.join(_gpkg_placeholders(record, gpkg_funcs))});
+        """
+
+    dbcur.execute(sql, tuple(record.values()))
+    return rowcount(dbcur)
+
+
+def _gpkg_placeholders(record, gpkg_funcs=None):
+    """Returns '?,?,?,?...' - where the nunber of ? returned is len(record)"""
```

Comment:
docstring says it returns a string but it actually returns an array

Reply:
Done